### ISSUE TITLE
Is it SQL injection safe?

### ISSUE THREAD

**Issue by xZero707** [joshcam/PHP-MySQLi-Database-Class#234]
I wonder if this class does filtering it self, or it still need to be done by the user?
I know that it uses prepared statements which are best in line fighting SQL injection, but I think that it alone is maybe not enough.

Few issues I had so far on PHP 5.6 (Linux Debian 7):
- Script which accepts user input and saves to database uses this class. 
When I entered large amount of data, one part of table got damaged, and another table got wiped clean. 
Script does accept only alphanumeric and those " ' . -  / ~ \ _ , ;
UPDATE: Repeating same after uploaded non-wiped backup of MySQL database passed successfully without damage to any other table.

<s> - I have one bigger query which fails without any errors. 
 Associative array is completely valid (not an multidimensional array)
 column names are valid.
I double checked.  What can be cause?
Tested with calls: "insert" and "update"
Where clause is ok.
</s> was a problem inside my program's else if.

Despite issues I experiencing, this class is just awesome, saved me many hours of work. 
Biggest thanks to even more awesome author for this. :+1:

**Comment by avbdr** [comment]
pretty hard to see where the problem is withou a data and a testcase. 
As for sql injections library is based on preprated statements so you should be safe if you will not be including data like where("a = ". $_GET['q']) or rawQuery ("select * from test where a = ". $_GET['q']).

**Comment by xZero707** [comment]
It's even difficult for me to investigate issue as it happened just once and I cannot reproduce it. Maybe it was problem with my server so database got corrupted, I can't know for sure.
What I did: on my local computer I build PHP script for supplying hundred of requests and sending them using CURL (POST) to server script which uses your class for saving data to database.
After local script finished, I checked wordpress website (same database) and found that is wiped out, like new installed wp. I checked database afterwards and found post and other tables wiped clean, and some tables got corrupted by loosing info about unique key.
--
I use "where" from user submitted data, but by preg_replace only alphanumeric characters are allowed, everything else is filtered out, so I guess it's safe. I expected rawQuery to be unsafe.
Thank you for your reply!
